### PR TITLE
Update timestamps.pyi

### LINKED ISSUE #797: "Timestamp" has no attribute "as_unit" [attr-defined]
I think we are missing the `.as_unit` method added to pd.Timestamp in 2.0

### PR DESCRIPTION
**Pull request by davetapley**
https://github.com/pandas-dev/pandas/commit/f3c46cd0899d5e11e0602798d9390c90e51e9ba7

https://github.com/pandas-dev/pandas/pull/48819

https://github.com/pandas-dev/pandas-stubs/issues/797

- [x] Closes #797 (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you add tests for this?

**Review by davetapley — COMMENTED** [review]
Hope this is sufficient 🙏🏻

File: pandas-stubs/_libs/tslibs/timestamps.pyi [hunk 1]
```diff
@@ -301,3 +301,6 @@ class Timestamp(datetime):
     def days_in_month(self) -> int: ...
     @property
     def daysinmonth(self) -> int: ...
+    @property
+    def unit(self) -> str: ...
+    def as_unit(self, unit: str, round_ok: bool = ...) -> Timestamp: ...
```
Comment (Dr-Irv) [inline pandas-stubs/_libs/tslibs/timestamps.pyi:306]: The only possible values for the return of `unit` are `Literal["s", "ms", "us", "ns"]`  In pandas, this is now defined in `pandas._typing.py` via:
```python
TimeUnit = Literal["s", "ms", "us", "ns"]
```
So can you do the following:
1. Add that line to `pandas-stubs/_typing.pyi` 
2. Make the return value of `unit` be `TimeUnit`
3. Make the allowable values of `unit` in `as_unit` to be `TimeUnit`

You probably will have to change the test for `ts.unit` to use `assert_type()` testing against `TimeUnit`

If you want to go beyond that, the declaration `Literal["s", "ms", "us", "ns"]` appears in the following files and should be replaced with `TimeUnit`:

- pandas-stubs/core/frame.pyi
- pandas-stubs/core/indexes/datetimes.pyi
- pandas-stubs/core/series.pyi
- pandas-stubs/io/json/_json.pyi

But if you don't want to make that change, it's OK - I can just create a new issue for that once this PR is done.

File: tests/test_scalars.py [hunk 2]
```diff
@@ -1530,6 +1531,8 @@ def test_timestamp_misc_methods() -> None:
         pd.Timestamp,
     )
 
+    check(assert_type(ts2.as_unit('ns'), pd.Timestamp), pd.Timestamp)
```
Comment (Dr-Irv) [inline tests/test_scalars.py:1534]:
```suggestion
    check(assert_type(ts2.as_unit("ns"), pd.Timestamp), pd.Timestamp)
```
use the pre-commit hook.  You need double quotes here.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by Dr-Irv** [comment]
Closing in favor of #864

**Pull request closed without merging**